lib/rubygems/remote_fetcher.rb

```diff
@@ -330,7 +330,10 @@ def cache_update_path uri, path = nil, update = true
     if update and path
       begin
         open(path, 'wb') do |io|
-          io.flock(File::LOCK_EX)
+          begin
+            io.flock(File::LOCK_EX)
+          rescue Errno::ENOTSUP
```

Comment:
@segiddins the issue I'm running into here is since it only has one class, if I use the approach that was used for `Gem.read_binary`, then on 1.8.7 it'll be equivalent to this:

```ruby
begin
  io.flock(File::LOCK_EX)
rescue
end
```

which seems... bad?

Any idea how to approach this?